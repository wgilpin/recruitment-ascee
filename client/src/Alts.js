import React from 'react';
import PropTypes from 'prop-types';
import Alt from './Alt';
import FetchData from './FetchData';

const propTypes = {
  onAltSelect: PropTypes.func,
  main: PropTypes.number,
};

const defaultProps = {
  main: null,
};

const styles = {
  outer: {
    height: '100%',
  },
  div: {
    display: 'block',
  },
  hr: {
    borderColor: '#555',
  },
};
export default class Alts extends React.Component {
  constructor(props) {
    super(props);
    this.state = { alts: props.alts };
  }

  handleClick = alt => {
    console.log('alts click ', this.props.onAltSelect.alt);
    this.setState({ selected: alt });
    if (this.props.onAltSelect) {
      this.props.onAltSelect(alt);
    }
  };

  componentDidMount() {
    let fetch = new FetchData({ id: this.props.main, scope: 'character/alts' });
    fetch
      .get()
      .then(data => {
        console.log(data);
        if (data.error === 'login') {
          window.location = '/app';
          return;
        }
        this.setState({ alts: data.info });
      })
      .catch(err => {
        console.log(err);
      });
  }

  render() {
    return (
      <div style={styles.outer}>
        <hr style={styles.hr} />
        {Object.keys(this.state.alts || {}).map(key => {
          const alt = this.state.alts[key];
          return (
            <Alt
              style={styles.div}
              name={alt.name}
              id={key}
              selected={this.state.selected === key}
              onClick={this.handleClick}
            />
          );
        })}
        <hr style={styles.hr} />
      </div>
    );
  }
}

Alts.propTypes = propTypes;
Alts.defaultProps = defaultProps;
