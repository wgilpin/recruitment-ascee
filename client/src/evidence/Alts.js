import React from "react";
import PropTypes from "prop-types";
import Alt from "../common/Alt";
import FetchData from "../common/FetchData";

const propTypes = {
  onAltSelect: PropTypes.func,
  onCorpSelect: PropTypes.func,
  main: PropTypes.number,
  childrenTop: PropTypes.bool,
  highlightMain: PropTypes.bool,
  showPointer: PropTypes.bool,
  style: {}
};

const defaultProps = {
  main: null,
  childrenTop: false,
  highlightMain: false,
  showPointer: false
};

const styles = {
  outer: {
    height: "100%",
    position: "relative",
    textAlign: "left",
    padding: "8px"
  },
  div: {
    display: "block"
  },
  hr: {
    borderColor: "#555"
  },
  sectionTitle: {
    fontWeight: 600,
    color: "#01799A",
    marginBottom: "6px"
  },
  corporation: {
    maxWidth: "300px"
  },
  corpCEO: {
    cursor: "pointer",
    textDecoration: "underline",
    color: "#01799A",
    maxWidth: "300px"
  },
  secStatus: {
    fontWeight: 500
  },
  red: {
    color: "red",
    fontWeight: 500
  },
  summary: {
    backgroundColor: "#222",
    padding: "4px"
  }
};
export default class Alts extends React.Component {
  constructor(props) {
    super(props);
    this.state = { alts: props.alts };
  }

  handleClickAlt = altId => {
    this.loadCharacterSummary(altId);
    this.setState({ selected: altId });
    if (this.props.onAltSelect) {
      this.props.onAltSelect(altId);
    }
  };

  handleClickCorp = corpId => {
    const { corporations, corporationId, selected } = this.state;
    if (
      this.props.onCorpSelect &&
      (corporations[corporationId] || {}).ceo_id === parseInt(selected)
    ) {
      this.props.onCorpSelect(corpId);
    }
  };

  loadCharacterSummary = altId => {
    return new FetchData({ id: altId, scope: "character", param2: "summary" })
      .get()
      .then(({ info }) =>
        this.setState({
          corporationName: info.corporation_name,
          corporationId: info.corporation_id,
          alliance: info.alliance_name,
          secStatus: info.security_status,
          corpRedlisted: "corporation_name" in info.redlisted
        })
      );
  };

  componentDidMount() {
    new FetchData({ id: this.props.main, scope: "user/characters" })
      .get()
      .then(data => {
        let mainData;
        if (this.props.highlightMain) {
          mainData = data.info[this.props.main];
          delete data.info[this.props.main];
        }
        this.setState({ alts: data.info, main: mainData });
      })
      .catch(err => console.log(err));
    new FetchData({ id: this.props.main, scope: "user/corporations" })
      .get()
      .then(data => this.setState({ corporations: data.info }))
      .catch(err => console.log(err));
  }

  renderSummary() {
    const { corporations, selected, corporationId, corpRedlisted } = this.state;
    const isCEO =
      (corporations[corporationId] || {}).ceo_id === parseInt(selected);
    let corpStyle = isCEO ? styles.corpCEO : styles.corporation;
    if (corpRedlisted) {
      corpStyle = { ...corpStyle, ...styles.red };
    }
    return (
      <div style={styles.summary}>
        <div>
          <div style={styles.secStatus}>
            Sec Status: {Math.round(this.state.secStatus * 100) / 100}
          </div>
          <div>
            Corp {isCEO && "[CEO]"}&emsp;
            <span
              style={corpStyle}
              onClick={() => this.handleClickCorp(this.state.corporationId)}
            >
              {this.state.corporationName}
            </span>
          </div>
        </div>
      </div>
    );
  }

  render() {
    const hasAlts = Object.keys(this.state.alts || {}).length > 0;
    return (
      <div style={{ ...styles.outer, ...this.props.style }}>
        {this.props.childrenTop && this.props.children}
        <hr style={styles.hr} />
        {this.props.highlightMain &&
          this.state.main && [
            <div style={styles.sectionTitle}>Main</div>,
            <Alt
              style={styles.div}
              name={this.state.main.name}
              id={this.props.main}
              selected={this.state.selected === this.props.main}
              onClick={this.handleClickAlt}
              showPointer={this.props.showPointer}
            />,
            this.state.selected === this.props.main && this.renderSummary(),
            <hr style={styles.hr} />
          ]}
        {hasAlts && <div style={styles.sectionTitle}>Alts</div>}
        {Object.keys(this.state.alts || {}).map(key => {
          const alt = this.state.alts[key];
          return [
            <Alt
              style={styles.div}
              name={alt.name}
              id={key}
              selected={this.state.selected === key}
              onClick={this.handleClickAlt}
            />,
            this.state.selected === key && this.renderSummary()
          ];
        })}
        {!this.props.childrenTop && this.props.children}
        {hasAlts && <hr style={styles.hr} />}
      </div>
    );
  }
}

Alts.propTypes = propTypes;
Alts.defaultProps = defaultProps;
