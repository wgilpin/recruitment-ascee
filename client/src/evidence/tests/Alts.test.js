import React from 'react';
import { mount, shallow } from 'enzyme';
import renderer from 'react-test-renderer';
import Alts from '../Alts';

let mockFetch;

let serverState;

beforeEach(() => {
  window.alert = () => {};
  serverState = {
    character: {
      alliance_id: 1354830081,
      alliance_name: 'Goonswarm Federation',
      ancestry_id: 18,
      birthday: '2013-06-09T08:05:44+00:00',
      bloodline_id: 8,
      character_id: 93459541,
      character_name: 'Andrew Nighthawk',
      corporation_id: 98409330,
      corporation_name: 'Ascendance',
      current_application_id: 18,
      description:
        '<font size="12" color="#bfffffff"></font><font size="18" color="#bfffffff">No man truely owns anything..  <br>He is mearly borrowing it until something bigger than him takes it away.<br><br><br><br><br><br><br><br><br><br><br><br></font><font size="18" color="#ff007fff">Recruitment is currently open. <br>Join</font><font size="18" color="#bfffffff"> </font><font size="18" color="#ff6868e1"><a href="joinChannel:player_-71200765//None//None">Ascendance recruitment</a></font><font size="18" color="#bfffffff"> </font><font size="18" color="#ff007fff">and speak to one of our recruiters</font>',
      gender: 'male',
      race_id: 8,
      redlisted: [],
      security_status: 5.008887629879013,
    },
    'user/corporations': {
      '98503673': {
        ceo_id: 93908806,
        ceo_name: 'Midgely Urea',
        corporation_name: 'Winnits and Other Things',
      },
      '98555774': {
        ceo_id: 2113387072,
        ceo_name: 'Kate Flashheart',
        corporation_name: "Kate's Manservant Recruitment Agency",
      },
    },
    'user/characters': {
      '94399527': {
        corporation_id: 98409330,
        corporation_name: 'Ascendance',
        name: 'Semirhage Boan',
      },
      '93908806': {
        corporation_id: 98503673,
        corporation_name: 'Winnits and Other Things',
        name: 'The Flight Lieutenant',
      },
      '2114725334': {
        corporation_id: 98409330,
        corporation_name: 'Ascendance',
        name: 'Entropics Applicant',
      },
    },
  };

  mockFetch = jest.fn().mockImplementation((scope, params) => {
    if (!serverState[scope.scope]) {
      throw new Error('Invalid Scope');
    }
    return Promise.resolve({
      info: serverState[scope.scope],
    });
  });
});

describe('<Alts>', () => {
  it('matches snapshot', async () => {
    const adminConfig = mount(
      <Alts
        main={'93908806'}
        childrenTop
        highlightMain
        showPointer
        fetcher={mockFetch}
      />
    );
    const tree = renderer.create(adminConfig);
    await Promise.resolve();
    await Promise.resolve();
    await Promise.resolve();
    expect(tree.toJSON()).toMatchSnapshot();
  });
});
