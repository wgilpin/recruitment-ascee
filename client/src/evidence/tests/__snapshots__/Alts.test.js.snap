// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Alts> matches snapshot 1`] = `
<div
  style={
    Object {
      "height": "100%",
      "padding": "8px",
      "position": "relative",
      "textAlign": "left",
    }
  }
>
  <hr
    style={
      Object {
        "borderColor": "#555",
      }
    }
  />
  <div
    style={
      Object {
        "color": "#01799A",
        "fontWeight": 600,
        "marginBottom": "6px",
      }
    }
  >
    Alts
  </div>
  <div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": null,
          "display": "block",
          "padding": 8,
        }
      }
    >
      <div
        style={
          Object {
            "display": "table-row",
          }
        }
      >
        <div
          style={
            Object {
              "display": "table-cell",
              "paddingLeft": "6px",
              "paddingRight": "6px",
              "textAlign": "left",
            }
          }
        >
          <img
            alt="Avatar"
            onClick={[Function]}
            src="https://image.eveonline.com/Character/94399527_64.jpg"
            style={
              Object {
                "backgroundColor": "black",
                "borderRadius": "50%",
                "boxShadow": "1px 3px 1px #111",
                "height": 32,
                "width": 32,
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "display": "table-cell",
              "height": "24px",
              "marginBottom": "auto",
              "marginTop": "auto",
              "paddingLeft": "8px",
              "paddingRight": "6px",
              "size": 14,
              "textAlign": "left",
              "verticalAlign": "middle",
            }
          }
        >
          Semirhage Boan
        </div>
      </div>
    </div>
  </div>
  <div>
    <div
      onClick={[Function]}
      style={
        Object {
          "cursor": null,
          "display": "block",
          "padding": 8,
        }
      }
    >
      <div
        style={
          Object {
            "display": "table-row",
          }
        }
      >
        <div
          style={
            Object {
              "display": "table-cell",
              "paddingLeft": "6px",
              "paddingRight": "6px",
              "textAlign": "left",
            }
          }
        >
          <img
            alt="Avatar"
            onClick={[Function]}
            src="https://image.eveonline.com/Character/2114725334_64.jpg"
            style={
              Object {
                "backgroundColor": "black",
                "borderRadius": "50%",
                "boxShadow": "1px 3px 1px #111",
                "height": 32,
                "width": 32,
              }
            }
          />
        </div>
        <div
          style={
            Object {
              "display": "table-cell",
              "height": "24px",
              "marginBottom": "auto",
              "marginTop": "auto",
              "paddingLeft": "8px",
              "paddingRight": "6px",
              "size": 14,
              "textAlign": "left",
              "verticalAlign": "middle",
            }
          }
        >
          Entropics Applicant
        </div>
      </div>
    </div>
  </div>
  <hr
    style={
      Object {
        "borderColor": "#555",
      }
    }
  />
</div>
`;
